flow/record/packer.py

```diff
@@ -8,7 +8,7 @@
 from .utils import EventHandler, to_str
 
 # Override defaults for msgpack packb/unpackb
-packb = functools.partial(msgpack.packb, use_bin_type=True)
+packb = functools.partial(msgpack.packb, use_bin_type=True, unicode_errors="backslashreplace")
```

Comment:
Thanks for the explanation and small example case, however this change shows you use `unicode_errors="backslashreplace"`  did you mean to use `unicode_errors="surrogateescape"`?

Reply:
No, I meant `backslashreplace`. This will reveal the bytes as-is for inspection. Per PEP-383 (https://peps.python.org/pep-0383/) the surrogateescape-handler allows developers to smuggle bytes across systems to facilitate non-UTF-8 integration, this is not the intention of Dissect I assume. In Dissect a clear byte-as-is description of invalid sequences would be preferable I guess.

Reply:
Then, why not directly use `backslashreplace` as the input (taken from your example):

```python
import sys
from flow.record import RecordDescriptor, RecordStreamWriter

rd = RecordDescriptor("desc", [("string", "data")])
rs = RecordStreamWriter(sys.stdout.buffer)
d = b"Hello \xa7 World".decode("utf-8", errors="backslashreplace")
rs.write(rd(data=d))
```

Output:

```shell
(dissect) user@mbp ~> python3 /tmp/test.py | rdump -L
[reading from stdin]
--[ RECORD 1 ]--
           data = Hello \xa7 World
        _source = None
_classification = None
     _generated = 2023-01-23 15:27:47.592665
       _version = 1
```

Reply:
That is indeed exactly what I am going to do in that PR. However, I thought it would be nice to have flow.record not raise an exception as well, there can be sources that do not perform this transformation right?

Reply:
Well, the problem is that you can never tell what these bytes are... in theory they could be anything. Unless you want to limit the input of flow.record to "known tools", but then we need to communicate this to the users. Also, I thought flow.record was more like a general tool that could also accept data streams from other sources, but maybe this is not the case?

Reply:
I was under the assumption that we cannot be sure about the input, although we prescribe UTF-8. But maybe we just want to make this a requirement (I do not see this in the docs... but maybe I am overlooking something)

Reply:
The thing is that the fieldtype is a `str` so if from a user perspective when you are parsing bytes and want to save it as a `str` the user needs to do a `.decode()` on it. Or else you can just store it in a `bytes` fieldtype correct?

I think it’s the user’s responsibility to choose the unicode error handling. if you assign bytes to a `flow.record` string field it will currently just do `.decode()` without any error handling. We can decide to force this to “backslashreplace” for example. If this is the intent then it needs to be changed somewhere else and not in the packer code.

if we want to allow “surrogateescaped” unicode strings we need to patch the packer and unpacker. So that’s why i was asking what the intent was. It’s also possible i’m overlooking something, if that is the case let me know.

Reply:
To recap, the msgpack packer doesn't allow 
serialization and deserialization of “surrogateescaped” strings by default. Any other python unicode str works fine from what i can tell. 

if we want to serialize bytes as a string and we want to choose a default error handling for the user (so that it always works), we need to change this in the string fieldtype code.

Reply:
I can understand the reasoning that flow.record is not responsible for invalid UTF-8 byte sequences. That is fine by me. Might be a good idea to mention the valid UTF-8 requirement both in the rdump manual and the flow.record manual though. Otherwise I fear that this might become a recurring topic.

Reply:
Do you have an example of an invalid UTF-8 `str` that is not `surrogateescape`'ed?